File: September/2026-09-07/45/45-jump-game-ii.py
# 45-jump-game-ii.cpp;array;help;leetcode

class Solution:
    def jump(self, nums: List[int]) -> int:
        jumps = 0
        curr = 0
        far = 0

        for i in range(len(nums) - 1):
            far = max(far, i + nums[i])

            if i == curr:
                jumps += 1
                curr = far

        return jumps

File: September/2026-09-07/45/45-jump-game-ii.cpp
// 45-jump-game-ii.cpp;array;help;leetcode

class Solution {
public:
    int jump(vector<int>& nums) {
        int jumps = 0, curr = 0, far = 0;

        for(int i = 0; i < nums.size() - 1; i++)  {
            far = max(far, i + nums[i]);

            if(i == curr) {
                jumps++;
                curr = far;
            }
        }

        return jumps;
    }
};